src/pmem.c

```diff
@@ -692,11 +978,36 @@ memset_nodrain_normal(void *pmemdest, int c, size_t len)
 static void *
 memset_nodrain_movnt(void *pmemdest, int c, size_t len)
 {
+	int i;
+	char buf[CHUNK_SIZE];
+	void *dest1 = pmemdest;
+	size_t cnt = 0;
+	int num_chunks;
+
 	/* way too chatty for LOG level 3 */
 	LOG(15, "pmemdest %p c 0x%x len %zu", pmemdest, c, len);
 
-	/* XXX stub version */
-	return memset(pmemdest, c, len);
+	/*
+	 * Align initial address. From there on set data in CHUNK_SIZE
+	 * bytes.
+	 */
+	memset_nodrain_normal(buf, c, CHUNK_SIZE);
+	pmem_align(&dest1, buf, &cnt, UP);
+	len -= cnt;
+
+	/* 128 byte chunks */
+	num_chunks = len/CHUNK_SIZE;
+
+	for (i = 0; i < num_chunks; i++) {
+		nodrain_movnt(dest1, buf, CHUNK_SIZE);
```

Comment:
I believe, I made similar suggestion for the previous version of this patch, but I'll repeat - this is very far fro optimal implementation.
*nodrain_movnt()* is called in a loop for each 128B chunk. This introduces a huge overhead, as for each chunk, it will perform some checks on input arguments first, then it will execute exactly one iteration of a loop in lines 693-713, and finally some other checks to detect unaligned tail.
IMHO, it would be much better to simply inline the code from lines 693-713 here. Or at least, extract the mentioned code into the separate *static inline* function and call it here.